lib/services/API/country_institute_api.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:mobile_app/constants.dart';
+import 'package:mobile_app/models/failure_model.dart';
+import 'package:mobile_app/utils/api_utils.dart';
+
+abstract class CountryInstituteAPI {
+  static const String COUNTRY = 'country';
+  static const String EDUCATIONAL_INSTITUTE = 'educational institute';
+  Future<dynamic> getSuggestions(String query, String url);
+}
+
+class HttpCountryInstituteAPI implements CountryInstituteAPI {
+  List<dynamic> data;
+
+  @override
+  Future<dynamic> getSuggestions(String query, String value) async {
```

Comment:
The way this API is handled makes it harder to use some more typeahead fields. It's better to have different functions for Country and Institute.

Reply:
The way of handling Country and Institute API's is exactly the same. If I make two separate functions ,then I think it  would just be code duplication. That is the reason why I've compressed both of them into a single function.

Reply:
@Nitish145 can you please give an update on this

Reply:
@jaggu21 you can make separate functions for both but can keep the main logic in a private function. The individual functions can basically call the main function by passing the API URL.